src/java.desktop/unix/classes/sun/awt/screencast/TokenStorage.java

```diff
@@ -109,26 +107,33 @@ private static Path setupPath() {
             return null;
         }
 
-        Path path = Path.of(userHome, REL_NAME);
+        Path primaryPath = Path.of(userHome, REL_NAME);
+        Path secondaryPath = Path.of(userHome, REL_NAME_SECONDARY);
+
+        Path path = Files.isWritable(secondaryPath) && !Files.isWritable(primaryPath)
```

Comment:
Sure, then it chooses the primary path and creates the missing directories as expected.

The idea is to use the secondary path for compatibility reasons only if it exists and the primary path doesn't.

Reply:
What I don't see here, which I think I expected to see, is that if only the secondary path exists, we copy what's there to the primary path, so that next time we go straight to the primary path.

Reply:
Sure, updated.